compiler/dyno/lib/resolution/Resolver.cpp

```diff
@@ -1327,10 +1327,15 @@ QualifiedType Resolver::typeForId(const ID& id, bool localGenericToUnknown) {
     }
 
     if (ct) {
+      auto newDefaultsPolicy = defaultsPolicy;
```

Comment:
Since, as we discussed, this code is doing something non-trivial, please add a comment describing what it's doing & why it's important